### PR TITLE
Feature/hrms 23

### PR DESCRIPTION
**Pull request by vikramwable**
Createed Employee search API with pagination

### CODE REVIEW

File: src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      case EMPLOYEE_FIRST_NAME:
+        sortBy = "firstName";
+        break;
+      case JOINING_DATE:
+        sortBy = "joiningDate";
+        break;
+    }
+    return sortBy;
+  }
+}
```
Comment (ninadbb) [inline src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java]: There are a few code-smells suggested by sonar lint link missing default, incorrect package naming convention, etc.
Please have a look. The rest all looks fine from functionality point of view.

**Review by ninadbb — COMMENTED** [review]
Apart from sonar lint suggestions, the rest looks fine from my side.
Sonar lint suggestions can found by adding sonar lint plugin in Intellij.

**Review by nikhilshinde57 — APPROVED** [review]

File: src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java (hunk 1 above)
Comment (vikramwable) [reply src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java]: Done there were 86 such a suggestions addressed all of them also formatted all java files with new formatter

Comment (ninadbb) [reply src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java]: Thanks, That's quite a heavy lifting. 👍

**Review by ninadbb — APPROVED** [review]

**Pull request merged**